artemis-server/src/main/java/org/apache/activemq/artemis/core/config/Configuration.java

```diff
@@ -668,6 +668,25 @@ Configuration addDiscoveryGroupConfiguration(String key,
     */
    Configuration setJournalDirectory(String dir);
 
+   String getJournalHistory();
+
+   File getJournalHistoryLocation();
+
+   /**
+    * Sets the file system directory used to store historical backup journal.
+    */
+   Configuration setJournalHistory(String dir);
+
+   int getJournalHistoryMaxDays();
+
+   Configuration  setJournalHistoryMaxDays(int days);
```

Comment:
This is a neat feature. I think I would call it something else from a configuration perspective.

journalRetention or just retention

configured with a:
 
int retentionLimit > 0
enum retentionUnit (seconds/days/numFiles)

Reply:
Oooohhhh... I like the name retention!!! Really nice!!!!